collider/CMakeLists.txt

```diff
@@ -1,8 +1,21 @@
 project (collider)
 
-add_library(${PROJECT_NAME} src/collider.cu)
+add_library(${PROJECT_NAME} src/collider.cpp)
 
-set_property(TARGET ${PROJECT_NAME} PROPERTY CUDA_ARCHITECTURES 61)
+if(THRUST_BACKEND STREQUAL "CUDA")
+    set_source_files_properties(src/collider.cpp PROPERTIES LANGUAGE CUDA)
```

Comment:
I guess this is the magic line that lets us avoid `.cu`? I wonder if we should rename the `.cuh` files to `.h` as well?

Reply:
Yes, we probably should rename the `.cuh` files to `.h`.